terraform/main.tf

```diff
@@ -0,0 +1,415 @@
+      //subnets_id = "subnet-0315baa5c81ff7d10"
+      subnet_ids    = module.vpc.private_subnets
+      capacity_type = "ON_DEMAND"
+      disk_size     = 30
+      k8s_labels = {
+        "testground.node.role.infra" = "true"
+      }
+
+      # Node Group scaling configuration
+      max_size     = 10
+      min_size     = 3
+      desired_size = 3
+
+      create_launch_template = true
+      kubelet_extra_args     = "--use-max-pods=false --max-pods=58"
+
+      //enable_bootstrap_user_data = true
+      //bootstrap_extra_args       = "--container-runtime dockerd"
+      //pre_bootstrap_user_data    = <<-EOT
+      //#!/bin/bash
+      //set -ex
+
+      //# https://docs.aws.amazon.com/eks/latest/userguide/choosing-instance-type.html#determine-max-pods
+      //# https://github.com/awslabs/amazon-eks-ami/blob/master/files/bootstrap.sh
+      //cat <<-EOF > /etc/profile.d/bootstrap.sh
+      //  export DEFAULT_CONTAINER_RUNTIME="dockerd"
+      //  export CONTAINER_RUNTIME="dockerd"
+      //  export USE_MAX_PODS=false
+      //EOF
+      //# Source extra environment variables in bootstrap script
+      //sed -i '/^set -o errexit/a\\nsource /etc/profile.d/bootstrap.sh' /etc/eks/bootstrap.sh
+      //sed -i 's/KUBELET_EXTRA_ARGS=$2/KUBELET_EXTRA_ARGS="$2 $KUBELET_EXTRA_ARGS"/' /etc/eks/bootstrap.sh
+      //EOT
+
+      //bootstrap_extra_args   = ""
+      //bootstrap_extra_args   = "--use-max-pods=false --max-pods=58 --container-runtime docker"
+      //bootstrap_extra_args   = ""
+      //pre_userdata = <<-EOT
+      //sudo bash -c 'cat <<SYSCTL > /etc/sysctl.d/999-testground.conf
+      //fs.file-max = 3178504
+      //net.core.somaxconn = 131072
+      //net.netfilter.nf_conntrack_max = 1048576
+      //net.core.netdev_max_backlog = 524288
+      //net.core.rmem_max = 16777216
+      //net.core.wmem_max = 16777216
+      //net.ipv4.tcp_rmem = 16384 131072 16777216
+      //net.ipv4.tcp_wmem = 16384 131072 16777216
+      //net.ipv4.tcp_mem = 262144 524288 1572864
+      //net.ipv4.tcp_max_syn_backlog = 131072
+      //net.ipv4.ip_local_port_range = 10000 65535
+      //net.ipv4.tcp_tw_reuse = 1
+      //net.ipv4.ip_forward = 1
+      //net.ipv4.conf.all.rp_filter = 0
+      //net.ipv4.neigh.default.gc_thresh2 = 4096
+      //net.ipv4.neigh.default.gc_thresh3 = 32768
+      //SYSCTL'
+      //sudo sysctl -p /etc/sysctl.d/999-testground.conf
+      //sudo bash -c 'cat <<LIMITS > /etc/security/limits.d/999-limits.conf
+      //* soft nproc 131072
+      //* hard nproc 262144
+      //* soft nofile 131072
+      //* hard nofile 262144
+      //LIMITS'
+      //EOT
+
+      //      iam:
+      // attachPolicyARNs:
+      // - arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryFullAccess
+      // - arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy
+      // - arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly
+      // - arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore
+      // - arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy
```

Comment:
can this be removed?

Reply:
I just uncommented it, I believe we do need it, the cluster works with and without it, but for preventing any issues, I'll keep it there